test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+	/*FIXME: fix for k8 v6 */
+	return "", nil
+}
+
+func (k *kubePod) getMACAddr(c *container, dev string) (string, error) {
+	out, err := k8sMaster.tbnode.RunCommandWithOutput(fmt.Sprintf("kubectl exec %s -- ip addr show dev %s | grep ether | head -1", c.containerID, dev))
+	if err != nil {
+		logrus.Errorf("Failed to get IP for container %q", c.containerID)
+		logrus.Println(out)
+	}
+
+	parts := regexp.MustCompile(`\s+`).Split(strings.TrimSpace(out), -1)
+	if len(parts) < 2 {
+		return "", fmt.Errorf("Invalid output from container %q: %s", c.containerID, out)
+	}
+
+	parts = strings.Split(parts[1], "/")
+	out = strings.TrimSpace(parts[0])
+	return out, err
+}
+
+/*
+* exec is used to run a specific command using kubectl on the host
+ */
+func (k *kubePod) exec(podName, args string, ns ...string) (string, error) {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+	cmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- ` + args
+	logrus.Debugf("Exec: Running command -- %s", cmd)
+	out, err := k8sMaster.runCommand(cmd)
+	if err != nil {
+		return out, err
+	}
+
+	return out, nil
+}
+
+/*
+* execBG executes a background process on the node
+ */
+func (k *kubePod) execBG(podName, args string, ns ...string) {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+	cmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- ` + args
+	logrus.Debugf("ExecBG: Running command -- %s", cmd)
+	k8sMaster.tbnode.RunCommandBackground(cmd)
+}
+
+/*
+* podExec function is used to run a command typically multiple commands
+* with pipes and redirections within the pod rather than the node
+ */
+func (k *kubePod) podExec(podName, args string, ns ...string) (string, error) {
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	// Backticks and quotes here ensure that the command
+	// is properly wrapped to execute on pod rather than on the node
+	podCmd := `sh -c '\''` + args + `'\''`
+	return k.exec(podName, podCmd, ns...)
+}
+
+/*
+* podExecBG function is used to run a background command
+* within the pod rather than the node
+ */
+func (k *kubePod) podExecBG(podName, args string, ns ...string) error {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	//  - Backticks and quotes here ensure that the command
+	//    is properly wrapped to execute on pod rather than on the node
+	//  - nohup along with & ensures that the process continues to live
+	//    after the shell is terminated
+	// Since the command is to run in BG in the pod and not on the node,
+	// RunCommandBackground is not required here
+
+	podCmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- nohup sh -c '\''` + args + ` &'\''`
+	logrus.Debugf("Pod Exec BG: Running command -- %s", podCmd)
+	return k8sMaster.tbnode.RunCommand(podCmd)
```

Comment:
So,podExec which uses k.exec is also run on k8sMaster. Just abstracted one more level. These are kubectl commands and needs to run on k8sMaster.